ext/nnstreamer/tensor_filter/tensor_filter_llama2.cc

```diff
@@ -0,0 +1,280 @@
+Tokenizer *TensorFilterLlama2c::tokenizer = nullptr;
+Sampler *TensorFilterLlama2c::sampler = nullptr;
+Config *TensorFilterLlama2c::config = nullptr;
+
+/**
+ * @brief Construct a new llama2c subplugin instance
+ */
+TensorFilterLlama2c::TensorFilterLlama2c ()
+{
+}
+
+/**
+ * @brief Destructor of TensorFilterLlama2c
+ */
+TensorFilterLlama2c::~TensorFilterLlama2c ()
+{
+}
+
+/**
+ * @brief Method to get an empty object
+ */
+tensor_filter_subplugin &
+TensorFilterLlama2c::getEmptyInstance ()
+{
+  return *(new TensorFilterLlama2c ());
+}
+
+/**
+ * @brief Configure llama2c instance
+ */
+void
+TensorFilterLlama2c::configure_instance (const GstTensorFilterProperties *prop)
+{
+  // build the Transformer via the model file
+  build_transformer (transformer, (char *) prop->model_files[0]);
+  config = &transformer->config;
+
+  if (steps == 0 || steps > config->seq_len)
+    steps = config->seq_len; // override to ~max length
+
+  // build the Tokenizer via the tokenizer file
+  build_tokenizer (tokenizer, (char *) prop->model_files[1], config->vocab_size);
+
+  // build the Sampler
+  build_sampler (sampler, config->vocab_size, TEMPERATURE, TOPP, RNG_SEED);
+}
+
+/**
+ * @brief Invoke llama2c using input tensors
+ */
+void
+TensorFilterLlama2c::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  char *result = NULL, *prompt, *output_data;
+  int len, string_idx = 0, i;
+
+  prompt = (char *) input[0].data;
+  len = strlen (prompt);
+
+  if (len > 1) {
+    prompt[strlen (prompt) - 1] = ' '; // Remove new line from text file
+  } else {
+    prompt = NULL;
+  }
+
+  result = get_tokens (transformer, tokenizer, sampler, prompt, steps);
+
+  /** @todo Make out_info flexible */
+  output[0].size = tokenizer->max_token_length * steps;
+  output_data = (char *) malloc (output[0].size);
+  len = strlen (result);
+  for (i = 0; i < len; i++) {
+    output_data[string_idx++] = result[i];
+  }
+  for (i = string_idx; i < (int) output[0].size; i++) {
+    output_data[i] = '\0';
+  }
+  output[0].data = output_data;
+
+  free (result);
+}
+
+/**
+ * @brief Get llama2c framework info.
+ */
+void
+TensorFilterLlama2c::getFrameworkInfo (GstTensorFilterFrameworkInfo &info)
+{
+  info.name = name;
+  info.allow_in_place = 0;
+  info.allocate_in_invoke = 1;
+  info.run_without_model = 0;
+  info.verify_model_path = 1;
+}
+
+/**
+ * @brief Get llama2c model info.
+ */
+int
+TensorFilterLlama2c::getModelInfo (
+    model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info)
+{
+  switch (ops) {
+    case GET_IN_OUT_INFO:
+      in_info.num_tensors = 1;
+      in_info.format = _NNS_TENSOR_FORMAT_FLEXIBLE;
+
+      /** @todo Make out_info flexible */
+      out_info.num_tensors = 1;
+      out_info.format = _NNS_TENSOR_FORMAT_STATIC;
+      out_info.info[0].type = _NNS_UINT8;
+      out_info.info[0].dimension[0] = tokenizer->max_token_length * steps;
+      break;
+    case SET_INPUT_INFO:
```

Comment:
Will you support changing tensor dimension later?
If not, return error for 'SET_INPUT_INFO' event.

Reply:
I will support output flexible tensors later, so I think `SET_INPUT_INFO` won't be supported
Updated it!
Thank you